packages/babel-plugin-ember-test-metadata/utils.js

```diff
@@ -56,48 +56,73 @@ function getProjectConfiguration(project) {
   return parsedProjectConfiguration;
 }
 
-function _getParsedClassicFilepath(pathSegments, projectConfiguration) {
-  const projectNamePathSeparators = projectConfiguration.pkg.name.split(path.sep);
+function _getClassicPath(preformattedPath, projectName) {
+  const formattedPath = preformattedPath
+    .slice(preformattedPath.lastIndexOf(projectName), preformattedPath.length)
+    .replace(`${projectName}/`, '');
 
-  pathSegments.splice(
-    0,
-    pathSegments.indexOf(projectNamePathSeparators[0]) + projectNamePathSeparators.length
-  );
-
-  return pathSegments.join(path.sep);
+  return formattedPath;
 }
 
-function _getParsedEmbroiderFilepath(pathSegments) {
-  const RELATIVE_PATH_ROOT = 2;
+function _getClassicPathWithAddon(preformattedPath, projectName, addons) {
+  const pathSegments = preformattedPath.split(path.sep);
+  const addonNames = addons.paths.map(addonPath => addonPath.split(path.sep)[1]);
+  const addonName = addonNames.find(name => pathSegments.includes(name));
 
-  // A typical Embroider pathSegments array would look something like:
-  // ['private', 'var', 'some-hash', 'embroider', 'another-hash', 'tests', 'acceptance', ...]
-  // This strips everything from the first segment up to and including 'another-hash',
-  pathSegments.splice(0, pathSegments.lastIndexOf('embroider') + RELATIVE_PATH_ROOT);
+  if (pathSegments.includes(addonName)) {
+    pathSegments.splice(0, pathSegments.indexOf("ember-add-in-repo-tests") + 1);
+    return [...pathSegments].join(path.sep)
+  } else {
+    return _getClassicPath(preformattedPath, projectName);
+  }
+}
 
+function _getEmbroiderPath(pathSegments) {
```

Comment:
This signature differs from those above. We should try to keep them consistent, so let's pass the `filePath` in here as well and do the split in this function.

```suggestion
function _getRelativePathForEmbroider(filePath) {
```